src/masternodes/mn_checks.cpp

```diff
@@ -1616,6 +1629,35 @@ class CCustomTxApplyVisitor : public CCustomTxVisitor
         return !res ? res : mnview.ICXCloseMakeOfferTx(*offer, CICXMakeOffer::STATUS_CLOSED);
     }
 
+    Res operator()(const CLoanSetCollateralTokenMessage& obj) const {
+        auto res = CheckICXTx();
```

Comment:
It is possible that there is no change output, a dust amount of change gets added to the fee. Looking at CheckICXTx I think it in general should check that there are no more than 2 outputs and not 2 exactly, however we cannot change that behaviour without putting it behind a fork, post-fork it could then be simplified.

Reply:
Yes, I mentioned earlier that we shouldn't check for exact number of vouts as it could be no change. But this can also be fixed (I used this approach before) with creating a change vout with 0 amount. This way we don't have to wait for new hard fork, and we also have this rigid check that someone cannot make a different tx than we expect...

Reply:
Change output below the dust threshold will be a non-standard TX and should be rejected on mainnet.

https://github.com/DeFiCh/ain/blob/20bd608730116a5683fde678cb1a5380c05681c6/src/policy/policy.cpp#L126

Reply:
Ok then. Yeah, in some previous implementation, my txes were special type so it wasn't considered as dust. We'll then check for 2 or less vouts on next fork.